msgs/include/simple_msgs/image.hpp

```diff
@@ -113,12 +113,22 @@ class Image : public GenericMessage
 
   bool operator==(const Image& rhs) const
   {
-    return ((header_ == rhs.header_) && (origin_ == rhs.origin_) && (encoding_ == rhs.encoding_) &&
-            (resX_ == rhs.resX_) && (resY_ == rhs.resY_) && (resZ_ == rhs.resZ_) && (width_ == rhs.width_) &&
-            (height_ == rhs.height_) && (depth_ == rhs.depth_) &&
-            (memcmp(data_.get(), rhs.data_.get(), data_size_) == 0) && (data_size_ == rhs.data_size_) &&
-            (num_channels_ == rhs.num_channels_));
+    if (data_ && rhs.data_)
```

Comment:
Check all the other members once, store the result, then check the image data (if present) and compare the two checks.